torch/_decomp/decompositions.py

```diff
@@ -4025,6 +4025,24 @@ def trunc(self: Tensor, **kwargs) -> Tensor:
     return torch.where(self > 0, torch.floor(self), torch.ceil(self))
 
 
+@register_decomposition([aten.div.Tensor_mode])
+def div_tensor_mode(
+    self: Tensor, other: Tensor, *, rounding_mode: Optional[str]
+) -> Tensor:
+    # aten::div.Tensor_mode(Tensor self, Tensor other, *, str? rounding_mode) -> Tensor
+    quotient = torch.div(self, other)
+    if rounding_mode is None:
+        return quotient
+    elif rounding_mode == "trunc":
+        return torch.trunc(quotient)
+    elif rounding_mode == "floor":
+        return torch.floor(quotient)
```

Comment:
This decomposition is wrong. Clearly it doesn't handle integer dtypes at all, but it also isn't accurate to the python floor div behavior which we carefully recreate in eager. The reference implementation gets this correct, so how about basing the decomposition on that?

Reply:
ah, you are correct. Apologies for the oversight.

Actually based on your feedback on the forums post, I'm thinking it may be more appropriate to make floor_divide a "core" operator; from the reference implementation a potential decomposition seems pretty complicated.

So I think it will be preferable to just `return torch.floor_divide(a, b)` for the `"floor"` mode instead of implementing the entire decomposition.